odc/stac/_mdtools.py

```diff
@@ -153,31 +153,20 @@ def has_proj_ext(item: Union[pystac.item.Item, pystac.collection.Collection]) ->
     """
     Check if STAC Item or Collection has projection extension.
 
-    :returns: ``True`` if PROJ exetension is enabled
+    :returns: ``True`` if PROJ extension is enabled
     :returns: ``False`` if no PROJ extension was found
     """
-    try:
-        ProjectionExtension.validate_has_extension(item, add_if_missing=False)
-        return True
-    except pystac.errors.ExtensionNotImplemented:
-        return False
+    return ProjectionExtension.has_extension(item)
 
 
 def has_raster_ext(item: Union[pystac.item.Item, pystac.collection.Collection]) -> bool:
     """
-    Check if STAC Item/Collection have EOExtension.
+    Check if STAC Item/Collection have RasterExtension.
 
-    :returns: ``True`` if Raster exetension is enabled
-    :returns: ``False`` if no Rasetr extension was found
+    :returns: ``True`` if Raster extension is enabled
+    :returns: ``False`` if no Raster extension was found
     """
-    try:
-        RasterExtension.validate_has_extension(item, add_if_missing=False)
-        return True
-    except pystac.errors.ExtensionNotImplemented:
-        return any(
-            ext_name.startswith("https://stac-extensions.github.io/raster/")
-            for ext_name in item.stac_extensions
-        )
+    return RasterExtension.has_extension(item)
```

Comment:
Tests are failing because of this change, I realize that newer version of `pystac` are more lenient as far as versions of extensions are concerned, but we might need to support older versions for a bit, so can we have that "dodgy check" back?

Reply:
On second thought, the "dodgy check" is kind of separate. I can add it back in.